.github/workflows/connector_checklist.yml

```diff
@@ -0,0 +1,23 @@
+name: Add Connector Merge Checklist
+on:
+  pull_request:
+    types: [opened]
```

Comment:
Note: if someone were to re-open an existing (right now) closed connector PR after this is merged, we wouldn't get a new checklist. I think the chances of that happening are probably less than someone accidentally or purposefully closing and reopening a connector PR after this is implemented, and since we don't want them to have to deal with duplicate checklists, I think this is fine, but wanted to note it